### DISCUSSION TITLE
I would like to test shadow root / web component

### DISCUSSION THREAD

**Discussion by quentinlebars** [vitest-dev/vitest#2454] [Q&A]
Hi,

I would like to test shadow root / web component with vitest, i didn't see any doc about this subject

Have you already do shadow root / web component testing ?

My project include web component and i want to test my component, it includes web component

Thanks in advance

**Comment by Corbie-42** [comment]
I need to test the shadow-root elements of my components too. Did you figure out how to do that?

**Comment by sofuxro** [comment]
I was able to test **shadow root / web component** using `element` and then access it's interface

``` ts
it('Testing email', async() => {
  const component = await mountSuspended(Form)

  component.find('nord-input').element.value = 'test@email.com'
  await component.find('nord-input').trigger('input')

  console.log(component.find('nord-input').element.value) // 'test@email.com'
  console.log('VM', component.vm.form.value.email)        // 'test@email.com'
})
```
